OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -220,12 +220,14 @@ void SanityCheckPath(Mobile mobile)
 
 			var nextCell = path[path.Count - 1];
 
+			var containsTemporaryBlocker = WorldUtils.ContainsTemporaryBlocker(self.World, nextCell, self);
+
 			// Next cell in the move is blocked by another actor
-			if (!mobile.CanMoveFreelyInto(nextCell, ignoredActor, true))
+			if (!mobile.CanMoveFreelyInto(nextCell, ignoredActor, true) || containsTemporaryBlocker)
```

Comment:
The chance that the cell actually contains a temporary blocker is quite slim. Even in that case it wouldn't save you anything that much.